ITIS_HW_1_BULLY.playground/Contents.swift

```diff
@@ -0,0 +1,746 @@
+    func AttackJock(student: Jock)
+    
+    func lookForLoot()
+}
+
+protocol Arena{
+    var fractionsForBattle: [Fraction] {get set}
+    func startBattle(fractions: [Fraction])
+}
+
+
+class Bullies: Fraction{
+    var name: String
+    var aliveStudents: [Student]
+    var deadStudents: [Student]
+    
+    init(name: String, students: [Student]){
+        self.name = name
+        self.aliveStudents = students
+        deadStudents = []
+    }
+    
+    func isDead() -> Bool{
+        if aliveStudents.isEmpty{
+            return true
+        }
+        return false
+    }
+    
+    
+    func Attack(attackedFraction: Fraction){
+        for attackingStudent in self.aliveStudents{
+            let attackedStudentNumber = Int.random(in: 0..<attackedFraction.aliveStudents.count)
+            print("Студент из \(self.name) атакует Студента из \(attackedFraction.name)...")
+            
+            switch attackedFraction.name{
+            case "Bullies":
+                attackingStudent.AttackBully(student: attackedFraction.aliveStudents[attackedStudentNumber] as! Bully)
+                break
+            case "Nerds":
+                attackingStudent.AttackNerd(student: attackedFraction.aliveStudents[attackedStudentNumber] as! Nerd)
+                break
+            case "Punks":
+                attackingStudent.AttackPunk(student: attackedFraction.aliveStudents[attackedStudentNumber] as! Punk)
+                break
+            case "Jocks":
+                attackingStudent.AttackJock(student: attackedFraction.aliveStudents[attackedStudentNumber] as! Jock)
+                break
+            default:
+                break
+
+            }
+            
+            
+            if attackedFraction.aliveStudents[attackedStudentNumber].isDead(){
+                print("Для студента из \(attackedFraction.name) это был критический урон!")
+            }
+            
+            attackedFraction.updateInfoAboutStudents()
+            if attackedFraction.isDead(){
+                break
+            }
+        }
+    }
+    
+    func lookForLoot(){
+        print("\(self.name):")
+        for player in self.aliveStudents{
+            player.lookForLoot()
+        }
+    }
+    
+    func updateInfoAboutStudents(){
+        var studentCounter = 0
+        while studentCounter < self.aliveStudents.count{
+            if self.aliveStudents[studentCounter].isDead(){
+                self.deadStudents.append(self.aliveStudents[studentCounter])
+                self.aliveStudents.remove(at: studentCounter)
+                studentCounter -= 1
+            }
+            studentCounter += 1
+        }
+    }
+    
+    func reviveAllStudents(){
+        for student in self.deadStudents{
+            self.aliveStudents.append(student)
+            deadStudents.removeAll()
+        }
+    }
+}
+
+class Bully: Student{
+    var hp: Int = 12
+    var hpMax: Int = 12
+    var damage: Int = 6
+    var fraction: String = "Bullies"
```

Comment:
```suggestion
    let fraction: String = "Bullies"
```
если свойство не изменяемое делаем его let